lib/src/line_to_polygon.dart

```diff
@@ -0,0 +1,181 @@
+  if (lines is FeatureCollection) {
+    featureEach(
+      lines,
+      (currentFeature, index) {
+        if (currentFeature.geometry is! LineString &&
+            currentFeature.geometry is! MultiLineString) {
+          throw exc;
+        }
+      },
+    );
+    List<List<Position>> list = [];
+    geomEach(
+      lines,
+      (
+        GeometryType? currentGeometry,
+        int? featureIndex,
+        Map<String, dynamic>? featureProperties,
+        BBox? featureBBox,
+        dynamic featureId,
+      ) {
+        if (currentGeometry is LineString) {
+          list.add(currentGeometry.coordinates);
+        } else {
+          list = [...list, ...currentGeometry?.coordinates];
+        }
+      },
+    );
+
+    lines = FeatureCollection<MultiLineString>(features: [])
+      ..features.add(Feature(geometry: MultiLineString(coordinates: list)));
+  } else if (lines is Feature) {
+    if (lines.geometry is LineString) {
+      lines = Feature<LineString>(
+        geometry: lines.geometry as LineString,
+        properties: lines.properties,
+        id: lines.id,
+      );
+    } else if (lines.geometry is MultiLineString) {
+      lines = Feature<MultiLineString>(
+        geometry: lines.geometry as MultiLineString,
+        properties: lines.properties,
+        id: lines.id,
+      );
+    } else {
+      throw exc;
+    }
+  } else if (lines is LineString) {
+    lines = Feature<LineString>(geometry: lines);
+  } else if (lines is MultiLineString) {
+    lines = Feature<MultiLineString>(geometry: lines);
+  } else {
+    throw exc;
+  }
+  if (!mutate) {
+    lines = lines.clone();
+  }
+
+  if (lines is FeatureCollection) {
+    List<List<List<Position>>> coords = [];
+    featureEach(
+      lines,
+      ((line, featureIndex) => coords.add(getCoords(lineStringToPolygon(
+              line, autoComplete, orderCoords, properties: {}))
+          as List<List<Position>>)),
+    );
+    return Feature(
+        geometry: MultiPolygon(coordinates: coords), properties: properties);
+  } else {
+    return lineStringToPolygon(lines, autoComplete, orderCoords,
+        properties: properties);
+  }
+}
+
+/// Converts LineString to Polygon
+/// Takes a optional bool autoComplete=true that auto completes linestrings
+/// Takes an optional orderCoords=true that sorts linestrings to place outer
+/// ring at the first position of the coordinates.
+Feature<Polygon> lineStringToPolygon(
+    GeoJSONObject line, bool autoComplete, bool orderCoords,
+    {Map<String, dynamic>? properties}) {
+  properties = properties ?? (line is Feature ? line.properties ?? {} : {});
+  var geom = line is LineString ? line : (line as Feature).geometry;
+  List<dynamic> coords = (geom is LineString || geom is MultiLineString)
+      ? (geom is LineString)
+          ? geom.coordinates
+          : (geom as MultiLineString).coordinates
+      : ((geom as Feature).geometry as GeometryType).coordinates;
```

Comment:
Please clean that up. The expression is super big and painful to read..

it should be sufficient to have a single if statement:
```dart
if(geom is Feature && geom.geometry is GeometryType) {
    geom = geom.geometry;
}
```

The rest should be dealt with in the if-else if-else clauses below